### PR TITLE
Add lasso selection example

### PR DESCRIPTION
**Pull request by DanAurea**
The example shall allow anyone requiring to use VisPy for lasso selection to be able to have a code reference with a rather complete implementation.

### CODE REVIEW

**Review by djhoese — CHANGES_REQUESTED** [review]
This is a really good start. Thanks for putting this all together. I had a couple comments and suggestions inline. I know there are a couple TODOs so there is still work to do but I wanted to give you feedback.

Some other thoughts: This should maybe be moved to the `scene` examples as it doesn't use the plotting API even though your use case is plotting-adjacent. This tool could be used for any use of Markers (or other visuals).

**Comment by DanAurea** [comment]
I pushed a cleaned up version of the example following your remarks + further extend optimization reusing matplotlib as a convenience method.
The code is much cleaner right now and easy to understand.

**Review by DanAurea — COMMENTED** [review]
Change have been performed accordingly + further optimization have been brought.

**Comment by djhoese** [comment]
This is on my TODO list, but due to other time sensitive things I can't promise I'll get to it immediately. Thanks for putting this all together. @brisvag if you have time and are interested feel free to also review.

File: examples/scene/lasso.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+Demonstrate the use of lasso selection.
+
+The lasso selection is done on a 2D scatter but could be extended further by user.
+"""
+import sys
+import numpy as np
+from vispy import app, scene
+from vispy.geometry import curves
+from vispy.scene import visuals
+from matplotlib import path
```
Comment (djhoese) [inline examples/scene/lasso.py:20]: It's too bad that this has to depend on matplotlib. It doesn't seem to fail which suggests we have matplotlib in our CI environment. Do we/can we have a fallback of the bounding box implementation if matplotlib isn't available? Perhaps something like:

```
import warnings

try:
    from matplotlib import path
except ImportError:
    warnings.warn("Lasso example requires matplotlib for more accurate selection. Falling back to naive bounding box selection.")
    path = None
```

And then check for `None` later in the code. If you could wrap the bounding box implementation in a path-like function/class that would be even more amazing but I realize this is a big ask.

**Review by djhoese — CHANGES_REQUESTED** [review]

**Comment by djhoese** [comment]
Forgot to mention in my review: This is amazing! I'd love to include this in my own application which already includes matplotlib. This example is so smooth. Nice job.

**Comment by DanAurea** [comment]
> Forgot to mention in my review: This is amazing! I'd love to include this in my own application which already includes matplotlib. This example is so smooth. Nice job.

I used matplotlib since the code used is one of the best compromise in term of performance but yes I could do a rework to add an algorithm without such dependency for sure.

I reworked a bit the code on my side to improve further performance in some situations but yes the lasso itself is very smooth.
I will propose a new commit without matplotlib or at least with a fallback solution.

**Comment by DanAurea** [comment]
I pushed a commit with fallback solution based on what I had found on Stackoverflow mixed with my bounding box prefiltering this seems to be as smooth as matplotlib based solution.

File: examples/scene/lasso.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+scatter = visuals.Markers()
+point_color = np.full((NUMBER_POINT, 4), FILTERED_COLOR)
+selected_mask = np.full(NUMBER_POINT, False)
+scatter.set_data(pos, edge_width=0, face_color=point_color, size=SCATTER_SIZE)
+
+view.add(scatter)
+
+def points_in_polygon(polygon, pts):
+    """
+    Get boolean mask of points in a polygon reusing matplotlib implementation.
```
Comment (djhoese) [inline examples/scene/lasso.py:66]:
```suggestion
    """Get boolean mask of points in a polygon reusing matplotlib implementation.
```

**Review by djhoese — APPROVED** [review]
This looks good to me. I'll give a day or so for @brisvag to provide any other feedback.

**Comment by brisvag** [comment]
Sorry for the delay, no comments on my side, looks awesome :)

**Comment by djhoese** [comment]
Thanks @DanAurea. This is great. I'm excited to fit this type of functionality into my own projects.

**Pull request merged**